storage/reports/reports_test.go

```diff
@@ -0,0 +1,680 @@
+package reports
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"reflect"
+	"testing"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/golang/mock/gomock"
+	"github.com/jinzhu/gorm"
+	"github.com/rs/zerolog"
+
+	"github.com/iryonetwork/wwm/reports"
+	"github.com/iryonetwork/wwm/storage/reports/db/mock"
+)
+
+var (
+	uuid1        = "45a1b7dd-dc9a-44b6-9c52-59a1489e3792"
+	uuid2        = "9f6e2d98-fde6-401e-8ef2-c33cd1038c51"
+	uuid3        = "afc1a438-0443-4cef-b99b-3d6f44e8f979"
+	uuid4        = "f6f4700e-bc7f-4ff7-9bcd-c139ec13fa2e"
+	uuid5        = "7cd54cf3-c01c-49eb-be23-8aae397701a3"
+	version1UUID = "74dc1c52-2993-4985-8940-faf61b329788"
+	version2UUID = "ff76b73e-03b5-415f-9123-5a0133513e33"
+	noErrors     = false
+	withErrors   = true
+	time1, _     = strfmt.ParseDateTime("2018-01-18T15:22:46.123Z")
+	time2, _     = strfmt.ParseDateTime("2018-01-19T18:33:57.123Z")
+	data         = "{\"it_is\": \"just_some_json_string\"}"
+
+	file1 = reports.File{
+		FileID:    uuid1,
+		Version:   version1UUID,
+		PatientID: uuid2,
+		CreatedAt: time1,
+		UpdatedAt: time1,
+		Data:      data,
+	}
+	file1V2 = reports.File{
+		FileID:    uuid1,
+		Version:   version2UUID,
+		PatientID: uuid2,
+		CreatedAt: time1,
+		UpdatedAt: time1,
+		Data:      data,
+	}
+	file2 = reports.File{
+		FileID:    uuid3,
+		Version:   version1UUID,
+		PatientID: uuid2,
+		CreatedAt: time2,
+		UpdatedAt: time2,
+		Data:      data,
+	}
+	file3 = reports.File{
+		FileID:    uuid4,
+		Version:   version1UUID,
+		PatientID: uuid5,
+		CreatedAt: time1,
+		UpdatedAt: time2,
+		Data:      data,
+	}
+)
+
+func TestInsert(t *testing.T) {
+	testCases := []struct {
+		title         string
+		fileID        string
+		version       string
+		patientID     string
+		timestamp     strfmt.DateTime
+		data          string
+		calls         func(*mock.MockDB)
+		errorExpected bool
+		exactError    error
+	}{
+		{
+			"Successfull insert",
+			uuid1,
+			version1UUID,
+			uuid2,
+			time2,
+			data,
+			func(db *mock.MockDB) {
+				gomock.InOrder(
+					db.EXPECT().Exec(
+						"INSERT INTO \"files\" (file_id, version, patient_id, created_at, updated_at, data) VALUES (?, ?, ?, ?, ?, ?) ON CONFLICT (file_id) DO UPDATE SET version = ?, updated_at = ?, data = ?",
+						uuid1,
+						version1UUID,
+						uuid2,
+						time2.String(),
+						time2.String(),
+						data,
+						version1UUID,
+						time2.String(),
+						data,
+					).Return(db),
+					db.EXPECT().GetError().Return(nil))
```

Comment:
invalid operation: db (variable of type *invalid type) has no field or method EXPECT